packages/browser-data-collector/src/collectors/performance-marks.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Internal dependencies
+ */
+import { getMarks } from '../api/user-timing';
+
+export const collector: Collector = ( report ) => {
+	const marks = getMarks();
+
+	marks.forEach( ( mark ) => {
+		// All data in the report is expected to be relative to `navigationStart`.
+		//
+		// mark.startTime is when the mark was craeted relative to the "time origin", which should
+		// be recorded in `performance.timeOrigin` but not all browsers supports it. However,
+		// `navigationStart` is reasonably close, therefore we don't need to correct `startTime`
+		report.data.set(
+			'mark__' + mark.name.toLowerCase().replace( /[^a-z]/g, '_' ),
```

Comment:
Should numbers and also be replaced with `_`?

Reply:
@sgomes `[^a-z]` will grab anything that's not in the `a-z` range, and that includes numbers.

Reply:
@tyxla Thanks, my comment wasn't too clear. The question is whether that was the intention 😄

Reply:
Ha, of course! Sorry for the misunderstanding. Now I'm also curious to see what's the rationale for keeping only lowercase Latin chars.

Reply:
My main reason is that this field will be parsed by ElasticSearch, so it has match its rules. I can't find any reliable source of what is valid as a field name, but, given the experience with ES so far, I rather be restrictive about allowed characters to make sure it doesn't drop the events.

Can you think of a good use case where we would need to add numbers (or symbols) to the mark name?

Reply:
My main concern is that folks add numbers to their marks as their only distinguishing text. For example: `signup_step_1`, `signup_step_2`. With this regex, they'll end up with a name collision: `signup_step__`.

This is something a developer can easily overcome by just using a different naming strategy, such as switching to spelled-out numbers, so definitely not a big problem in that regard. I'm mainly concerned they won't expect to have to, and won't easily be able to figure out the problem.

Reply:
Make sense, I'll allow numbers and do some tests with ElasticSearh

Reply:
related: I tried to add support for reporting arrays when two marks collide (after all, it is ok to have two marks with the same name) but Kibana is not very good at querying and visualising properties with array values.

So in case of collision, it will report the first mark.